### PR TITLE
Initial Commit

### PR DESCRIPTION
**Pull request by rodney528**
In difficultyArray you can put a whole new difficulty list which will fix some issues when changing songs doesn't want to fucking work! Because if that song doesn't have said chart/diff then nothing happens so PLZ MERGE THIS!!

### CODE REVIEW

**Review by charlesisfeline — APPROVED** [review]
but mistakes aside, this is, pretty nice

**Review by moxie-coder — APPROVED** [review]

**Comment by jimmyjamescool** [comment]
I HATE MAPS. I FUCKING WANNA BURN EVERY SINGLE MAP I SEE. YOU LOST? GET LOST. HAVE A DEGREE IN CARTOGRAPHY? GO FUCK YOURSELF. YOU WANNA BE A PIRATE AND SAIL THE SEVEN SEAS? GO TAKE A PIPE BOMB WITH YOU! LIKE BEING ABLE TO TAKE THE FASTEST ROUTE TO AND FROM WORK? I'M GOING TO KILL YOU. YOU FUCKING KNOW WHAT MAPS ARE? JUMP OFF A CLIFF! EVERY MAP LOVER, HATER, USER, AND THE LIKE; ALL SHALL MEET THEIR DOOM!

**Review by jimmyjamescool — APPROVED** [review]

**Review by jimmyjamescool — APPROVED** [review]

**Review by jimmyjamescool — APPROVED** [review]

**Review by jimmyjamescool — APPROVED** [review]

**Review by jimmyjamescool — APPROVED** [review]

**Review by jimmyjamescool — APPROVED** [review]

**Review by jimmyjamescool — APPROVED** [review]

**Review by jimmyjamescool — APPROVED** [review]

**Review by jimmyjamescool — APPROVED** [review]

**Comment by TehPuertoRicanSpartan** [comment]
what the fuck happened

**Review by charlesisfeline — APPROVED** [review]

**Review by charlesisfeline — APPROVED** [review]

**Review by charlesisfeline — APPROVED** [review]

**Comment by RodneyAnImaginativePerson** [comment]
> you forgot to change the pr name tho

I deleted the old one, thats why I kept the name the same.

File: source/psychlua/FunkinLua.hx [hunk 1]
```diff
@@ -419,11 +419,10 @@ class FunkinLua {
 			return false;
 		});
 
-		Lua_helper.add_callback(lua, "loadSong", function(?name:String = null, ?difficultyNum:Int = -1) {
-			if(name == null || name.length < 1)
-				name = PlayState.SONG.song;
-			if (difficultyNum == -1)
-				difficultyNum = PlayState.storyDifficulty;
+		Lua_helper.add_callback(lua, "loadSong", function(?name:String = null, ?difficultyNum:Int = -1, ?difficultyArray:Array<String> = null) {
+			if (difficultyArray != null) CoolUtil.difficulties = difficultyArray;
```
[the remark this answers is not in the document]
Comment (RodneyAnImaginativePerson) [reply source/psychlua/FunkinLua.hx:423]: Woops

[the remark this answers is not in the document]
Comment (RodneyAnImaginativePerson) [reply source/psychlua/FunkinLua.hx:423]: 0.6.3 habits

**Comment by RodneyAnImaginativePerson** [comment]
poop to bigAssShit lmao

**Pull request closed without merging**